contributed_definitions/nyaml/NXxps.yaml

```diff
@@ -0,0 +1,451 @@
+category: application
+doc: |
+  This is the application definition for X-ray photoelectron spectroscopy.
+type: group
+NXxps(NXmpes):
+  (NXentry):
+    definition(NX_CHAR):
+      enumeration: [NXxps]
+    method:
+      exists: recommended
+      doc: |
+        A name of the experimental method according to `Clause 11`_ of
+        the `ISO 18115-1:2023`_ specification.
+        
+        Examples for XPS-related experiments include:
+            * X-ray photoelectron spectroscopy (XPS)
+            * angle-resolved X-ray photoelectron spectroscopy (ARXPS)
+            * ultraviolet photoelectron spectroscopy (UPS)
+            * hard X-ray photoemission spectroscopy (HAXPES)
+            * near ambient pressure X-ray photoelectron spectroscopy (NAPXPS)
+            * electron spectroscopy for chemical analysis (ESCA)
+        
+        .. _ISO 18115-1:2023: https://www.iso.org/standard/74811.html
+        .. _Clause 11: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:sec:11
+    xps_coordinate_system(NXcoordinate_system):
+      doc: |
+        Description of the XPS base coordinate system, which is defined such that the positive z-axis
+        points along the sample normal, and the x- and y-axes lie within the sample surface.
+        
+        The fields `x`, `y`, `z`, and `scaling` are used to describe the orientation of the XPS
+        coordinate system with respect to the beam coordinate system (.).
+      origin:
+        enumeration: [sample surface]
+      handedness:
+        enumeration: [right_handed]
+      z_direction:
+        enumeration: [sample normal]
+      x(NX_NUMBER):
+      y(NX_NUMBER):
+      z(NX_NUMBER):
+      offset(NX_NUMBER):
+    (NXinstrument):
+      doc: |
+        XPS spectrometer.
+      (NXelectronanalyser):
+        depends_on:
+          exists: recommended
+          doc: |
+            Reference to the transformation describing the orientation of the analyzer
+            relative to the beam.
+          enumeration: [transformations/angle_beam_analyser]
+        transformations(NXtransformations):
+          angle_beam_analyser(NX_NUMBER):
+            exists: recommended
+            unit: NX_ANGLE
+            doc: |
+              Angle between incoming beam and the center analyser axis.
+            \@transformation_type:
+              enumeration: [rotation]
+            \@depends_on:
+              doc: |
+                Should be a link to the XPS base coordinate system, e.g.
+                /entry/instrument/xps_coordinate_system.
+      (NXenergydispersion):
+        scheme:
+          enumeration: [hemispherical, double hemispherical, cylindrical mirror]
```

Comment:
What about tof, etc.? I would not restrict here at all.

Reply:
ok